input/pagecontent/algorithms.md

```diff
@@ -1,257 +1,189 @@
+* **Converting to Age:** Dates can be converted into an age at the time of an event. This age can then be generalized (e.g., grouped into 5-year brackets) to further reduce risk.
 
-PID\|123\|123\|123456\|123\|FamilyName^""^GivenName^""^""^""\|\|19900113\|U\|Alias\|U\|""^""^""^^840??^""\|US\|HomePh\|BusPh\|U\|U\|U\|PSEUDO1234\|U\|U\|U\|U\|U\|""
+##### De-Identifying Geographical Locations and Geolocation Data
 
-```
+Geographic data is highly identifying. Techniques for de-identifying it include:
 
-#### Other Considerations
+* **Generalization:** Replace precise addresses with broader regions, such as zip codes, counties, or states. The level of generalization depends on the population density of the area.
+* **Geometric Aggregation:** Combine geographic points into larger polygons, obscuring the exact location of individuals.
+* **Suppression:** Remove location data entirely for individuals in sparsely populated areas where they could be easily identified.
 
-Care must be taken to ensure redacted data remain syntactically correct. A HL7 CDA document that complies with a template has specific rules regarding discrete data elements and required structure. Post processing after de-identification may be required to create a document that still complies with the template. Template designers may need to consider the needs of de-identification in the design of new templates.
+##### De-Identifying Genomic Information
 
-### Fuzzing
+Genomic data is unique and inherently identifying. De-identification is extremely challenging and often focuses on risk mitigation rather than complete anonymization. Key strategies include:
 
-#### Description
+* **Controlled Access:** Restricting access to genomic data to authorized researchers under strict data use agreements.
+* **Data Partitioning:** Splitting genomic data from demographic and clinical data, with access controls on both.
+* **Cryptographic Methods:** Using techniques like homomorphic encryption to allow computation on encrypted genomic data without decrypting it.
 
-Fuzzing adds apparently random modifications to data while remaining within certain constraints. For example a random amount of time can be added to or removed from person’s birth date. The goal of fuzzing is to remove as much accuracy as possible while still meeting the intended use. The design phase should determine the accuracy that must be preserved. Fuzzing is the only de-identification approach that provides control over the statistical characteristics of the data.
+##### De-Identifying Text Narratives and Qualitative Information
 
-#### Applicability
+Free-text fields like clinical notes or physician comments are a major challenge for de-identification because they often contain direct and indirect identifiers in unstructured formats. Automated de-identification of this content requires a combination of techniques.
 
```

Comment:
the value of a free-text field is questionable. If it is not structured, then it is not helpful to analysis. We should point this out. I often find that people tend to want to keep the text fields but don't know why. When they realize how useless they are, they realize that redaction is fine.

Failing to remove some direct or indirect identifier is highly likely.